modules/app/src/main/java/org/locationtech/jtstest/testbuilder/io/shapefile/PolygonHandler.java

```diff
@@ -299,12 +301,15 @@ public int getShapeType(){
     }
     public int getLength(Geometry geometry){
         
-           MultiPolygon multi;
+          Polygon[] polygonArray;
+          MultiPolygon multi;
         if(geometry instanceof MultiPolygon){
             multi = (MultiPolygon)geometry;
         }
         else{
-            multi = new MultiPolygon(new Polygon[]{(Polygon)geometry},geometry.getPrecisionModel(),geometry.getSRID());
+            polygonArray = new Polygon[]{(Polygon)geometry};
```

Comment:
Actually since Geometry now implements getNumGeometries and getGeometryN, there is no need to create a temporary MultiPolygon any more.

Reply:
@dr-jts Thanks for looking over this, f9f323a removes the multipolygon construction and calls geometry directly!